addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PromotionManager.java

```diff
@@ -765,8 +766,9 @@ private PathsPromoteResult runPathPromotions( final PathsPromoteRequest request,
         Set<Group> affectedGroups;
         try
         {
-            affectedGroups = storeManager.query().getGroupsAffectedBy( targetKey );
-            logger.info( "Target {} affected groups: {}", targetKey, affectedGroups );
+            String filter = config.getAffectedGroupsExcludeFilter();
+            affectedGroups = promotionHelper.filterAffectedGroups( storeManager.query().getGroupsAffectedBy( targetKey ), filter );
```

Comment:
Just to be clear, this will not change the processing in StoreContentListener, which clears metadata when a hosted repo is added to a group...at least, I don't think it will. That gets triggered off of the ArtifactStoreUpdateEvent, IIRC. So if that's true, it wouldn't save any performance on the temporary build use case we're working with now.

Do we need to handle this type of logic there too?

Reply:
This only affects by-path promotion atm. To use it in adding-hosted-to-group case, we need to do it in a more generic way. I think we can do it. Let's leave this open for a while. I will come back to it after fixing the tricky stage testing issue.